Sources/NIOCore/EventLoop.swift

```diff
@@ -19,15 +19,28 @@ import Dispatch
 ///
 /// A `Scheduled` allows the user to either `cancel()` the execution of the scheduled task (if possible) or obtain a reference to the `EventLoopFuture` that
 /// will be notified once the execution is complete.
-public struct Scheduled<T> {
+public struct Scheduled<T>: NIOSendable {
```

Comment:
Does this need to be `preconcurrency`? We can only be `Sendable` where the generic type is `Sendable`.

Reply:
Good question and I think the question boils down to if `EventLoopPromise` should be unconditionality `Sendable` or not.

Reply:
I have made it only conditionally `Sendable`  as `EventLoopPromise` should have been only conditionally `Sendable` as well.